recipe/meta.yaml

```diff
@@ -1,311 +1,309 @@
-{% set version = "4.6.0" %}
-{% set build_num = "5" %}
+# This recipe creates 3 packages
+# libopencv, py-opencv, opencv
+#
+# libopencv actually contains all the files, including the site-packages
+# file needed to run opencv with python.
+#     opencv-dnn seems to change on every patch version, therefore, we pin
+#     to the patch version x.x.x
+#
+# py-opencv is a meta package that constrains the python dependencies
+#     This one is pinned to the minor version. x.x
+#
+# opencv is a backward compatible package, with no pinnings
+#
+# This approach is different than the one in defaults. Default's package
+# seperates well the files generated with python dependencies, from those
+# generated without. Unfortunately, this isn't entirely supported by
+# upstream. Their hacks are quite elaborate, making it seem difficult to
+# maintain as opencv has been changing their build system recently.
+#
+# By putting all the generated files in 1 package, this makes the build process
+# much easier, at the expense of a few MBs in the 'lib' package.
+{% set version = "4.10.0" %}
+{% set major_version = version.split('.')[0] %}
+{% set PY_VER_MAJOR = PY_VER.split('.')[0] %}
+{% set PY_VER_MINOR = PY_VER.split('.')[1] %}
+
+{% set build = 0 %}
 
-{% if target_platform != "linux-aarch64" and target_platform != "linux-64"%}
-{%   set enable_testingui = true %}
-{% else %}
-{%   set enable_testingui = false %}
-{% endif %}
 
-{% if build_variant == "headless" %}
-{%   set package_name = "opencv-python-headless" %}
-{% else %}
-{%   set package_name = "opencv" %}
+# Allow the qt version to be easily read from the build number 100's value.
+{% if qt_version == "5" %}
+{% set build = build + 500 %}
 {% endif %}
 
 package:
-  name: opencv-suite
+  name: libopencv
```

Comment:
looks like there's a top-level package that wants to be created  here, along with their own build strings and tests (libopencv) and then other packages created via the outputs (opencv, pyopencv). This doesn't seem right, unless I'm missing something. See here for more info: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section

Reply:
I think it's quite opposite and implicitly implemented by conda-build:
_When viewing the top-level package as a collection of smaller subpackages, it may be convenient to define the top-level package as a composition of several subpackages. **If you do this and you do not define a subpackage name that matches the top-level package/name**, conda-build creates a metapackage for you._

see https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#implicit-metapackages

IMO, conda-forge found a hacky way to build the package if `package/name` and `outputs/name[0]` are the same.

I'm not sure why but only this way succeeded on all platforms.

Reply:
ok thanks. So do you mean that conda-forge's way - this way - is kind of hacky? The way before worked ok AFAIU... I'm still a little unsure about why the structure of the recipe was changed so much, apart from to align with conda-forge, which is often a good idea but not if their recipe isn't that great... I could be wrong...

Reply:
I made many attempts to build it on `win64` using the old structure (see https://github.com/AnacondaRecipes/opencv-feedstock/pull/29) but without success. Other platforms worked fine.
Then I decided to rewrite the recipe based on the conda-forge recipe (dropping aside `headless` and `normal` variants temporarily). Something is wrong with CMake flags or sharing environments between subpackages.
To be honest it makes me crazy.
The package has already wasted a lot of my time and I still have a long way to go before completing the task.

Reply:
ok, understood. let's discuss over zoom.